### ISSUE TITLE
"Create missing overrides" and "Create field" fixes not available for enums

### ISSUE THREAD

**Issue by srawlins** [dart-lang/sdk#55264]
In this code:

```dart
abstract class C {
  void M();
}
enum E implements C {
  one, two;
}
```

The "create missing overrides" refactoring is not offered.

In this code:

```dart
enum E {
  one, two;
}
int f(E e) {
  return e.x;
}
```

The "create field" refactoring is not offered.

**Comment by bwilkerson** [comment]
Both would be good to have.

> The "create field" refactoring is not offered.

Should we also have a "create enum constant" fix?

**Comment by NBCBM** [comment]
The "create field" refactoring is not offered. Enums in Dart cannot have instance variables or fields. They are limited to declaring constant instances only.

To resolve these issues:

For the first scenario, if you need methods associated with enum instances, you may need to refactor your code to use a class hierarchy or a different approach altogether.
For the second scenario, if you need instance variables associated with enum instances, consider using a class instead of an enum.
Enums in Dart are primarily intended for representing a fixed set of constant values, rather than containing methods or instance variables.

**Comment by bwilkerson** [comment]
> Enums in Dart cannot have instance variables or fields.

The [`enhanced-enums`](https://github.com/dart-lang/language/blob/main/accepted/2.17/enhanced-enums/feature-specification.md) feature, which shipped in 2.17, makes it possible to have instance fields in an enum, though instance fields _are_ required to be `final` so that the instances can be constants.

But you raise an interesting question. We'd need to decide whether "create field" added an initializer (which isn't possible in all cases), made the field nullable, or added the field as a parameter to every constructor.
